### PR TITLE
Staff: Fixes for Certificates view

### PR DESCRIPTION
**Pull request by tfamula**
Hi @mjansenDatabay 
this is a PR which fixes multiple things in the Certificates view in Staff. Most of the bugs are mentioned in [36731](https://mantis.ilias.de/view.php?id=36731) and [36732](https://mantis.ilias.de/view.php?id=36732). 
The problems were in summary:
- Filter fields were partly not working
- Table column for email was not filled
- Table sortation was not working at all
- Table navigation was not working at all

Some of the code changes had to be done in the Certificate API for Staff, so please have a look.

I will pick this to release 8 as soon as PR is merged.

### CODE REVIEW

**Comment by mjansenDatabay** [comment]
Hi @tfamula,

thanks a alot for the changes in the certificate related files.

As the certificate API had been introduced I already had the feeling that the dependencies were modeled incorrectly. In discussions between the component maintainers of `MyStaff` (Martin) and `Certificates` (which included my former colleague, and of course myself) I already mentioned this, so we all were aware of this. But because of huge performance considerations regarding the `MyStaff` view (it is a typical table with potentially lot of rows) we decided to stick with this approach (just wrapping a SQL query).
Our assumption became true now: The repository had initially knowledge about the user and object domain, which was just okay because these are exactly the two dependencies of the certificates component. If this repository now also knows implementation details about the organizational units (means: database tables), then this is actually too much knowledge about foreign domains (and in terms of organizational units, we're not interested in this domain at all, from a certificate perspective). .
Summary: Your PR modifies certificate related files for filtering and ordering purposes in `MyStaff`, which is totally wrong IMO. This is not your fault, you just tried to fix the issues by adapting the given mechanisms.

I will not block this PR, but I would like to hear your opinion about this. If you like we could also have a chat in Discord about this.

Best regards,
Michael

**Review by mjansenDatabay — APPROVED** [review]

**Comment by mjansenDatabay** [comment]
@tfamula You can also pick this to `release_8` and `release_7`, if required.

**Comment by tfamula** [comment]
Thanks @mjansenDatabay! I cherry-picked to release 8 and manually committed to release 7 (because of some differences in comparison to release 8/trunk).

**Pull request merged**
